### PR TITLE
Add channel resolver

### PR DESCRIPTION
**Pull request by impressor615**
## Description
- add Channel Resolver
- ~add jest global setup, global teardown for starting and closing server and syncing db.~
- change field name of schema ChannelInput `friendsId` to `friendIds`.

## Related Issues
#26

## Tests
run `yarn test` command

### CODE REVIEW

**Comment by impressor615** [comment]
I guess I need to migrate db models. But, I am not sure how I can do.
Could you help me to solve this problem? @hyochan

File: schemas/channel.graphql [hunk 1]
```diff
@@ -13,7 +13,7 @@ type Channel {
   type: ChannelType
   name: String
   messages: [Message]
-  membership: Membership
+  membership: [Membership]
```
Comment (hyochan) [inline schemas/channel.graphql]: This should be reverted. [Channel] doesn't have many [Membership].

Comment (hyochan) [reply schemas/channel.graphql]: Sorry I was confused. Yes, we should have to get many memberships from the [Channel] (we need to know who is in the channel).
However, it looks like we need additional queries for `membership`.
What about adding both `memberships` and `membership` or `memberships` and  `myMembership`?

File: src/resolvers/channel.ts [hunk 2]
```diff
@@ -38,21 +38,52 @@ const resolver: Resolvers = {
...
+          }),
+        };
+      });
+      await membershipModel.bulkCreate(membershipData);
+
       return channel;
     },
   },
+  Channel: {
+    membership: (_, args, { models }): Promise<Membership[]> => {
```
Comment (hyochan) [inline src/resolvers/channel.ts]: As explained above, `memberships` (plural) will be better naming in this case.

Comment (impressor615) [reply src/resolvers/channel.ts]: I agree with that. I applied it. Thanks for your review.

File: schemas/channel.graphql (hunk 1 above)
Comment (impressor615) [reply schemas/channel.graphql]: I changed membership to memberships (plural). However, I am confused about the usage of what you mentioned as `membership` or `myMembership`. Is it for querying the channel's owner membership? @hyochan

Comment (hyochan) [reply schemas/channel.graphql]: It's for querying my membership associated with `Channel`.
When you are querying `Channel`, you'd want to get `myMembership` together at one resolver!
Please tell me when it still isn't clear!

**Comment by hyochan** [comment]
@impressor615 I got a news for you in my [last commit](https://github.com/dooboolab/hackatalk-server/pull/31/commits/3503a11ed5fa570d3ac06d1a300151c27e9a9375). How do you think about handling jest global as I've provided?

File: schemas/channel.graphql (hunk 1 above)
Comment (impressor615) [reply schemas/channel.graphql]: Ah ha! I will work on it and push commits  as soon as possible. Thanks 🙏

**Comment by impressor615** [comment]
Your approach looks clearer and more readable than what I built before. Thanks. I will finish this PR based on the latest commits as soon as possible! @hyochan

File: schemas/channel.graphql (hunk 1 above)
Comment (impressor615) [reply schemas/channel.graphql]: I added `myMembership` fragment query resolver in Channel resolver. I wanted not to call `getUser` again in fragment query resolver. However, I could not figure out how I can remove the duplicated db query for now. Is there any other option to share data between resolver and fragment resolver?

Comment (hyochan) [reply schemas/channel.graphql]: @impressor615 It isn't a good idea to make more threads to resolved review. Let's move it down.

**Comment by hyochan** [comment]
@impressor615 Currently, I have no clue about this yet. If you can find this out for us, it'd be great (this is why I always work on open source 😀 ).

If you see my [new commit](https://github.com/dooboolab/hackatalk-server/pull/31/commits/9d9e2b5ea5f922b0acf1d2f89d5529b1df283bd5), I've added `verifyUser` function that don't need promise method and either extra query to resolve some sense of your issue which may be part of it (I guess).

**Comment by impressor615** [comment]
Sorry for the late reply. And, thanks for your help resolving the issues I had. 👍

**Pull request merged**